compiler/src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -626,17 +626,19 @@ object Erasure extends TypeTestsCasts{
 
                     if (isRequired) {
                       // check for clashes
-                      val clash: Option[Symbol] = oldSymbol.owner.info.decls.lookupAll(bridge.name).find {
-                        sym =>
-                          (sym.name eq bridge.name) && sym.info.widen =:= bridge.info.widen
-                      }.orElse(
-                        emittedBridges.find(stat => (stat.name == bridge.name) && stat.tpe.widen =:= bridge.info.widen)
+                      val clash: Option[Symbol] = oldSymbol.owner.info.memberExcluding(bridge.name, Flags.Bridge).alternatives.find {
+                        denot => {
+                          !denot.symbol.owner.is(Flags.Trait) && (denot.symbol.owner ne defn.ObjectClass) &&
+                          !denot.symbol.owner.derivesFrom(bridge.owner) && denot.symbol.info.widen =:= bridge.info.widen
+                        }
+                      }.map(_.symbol).orElse(
+                        emittedBridges.find(stat => (stat.name eq bridge.name) && stat.tpe.widen =:= bridge.info.widen)
```

Comment:
I'd prefer to change `eq` back to `==` to keep the diff minimal. This method has a complex history and it's important to keep it clean.